sdk/core/Azure.Core.Expressions.DataFactory/src/DataFactoryExpressionJsonConverter.cs

```diff
@@ -2,41 +2,238 @@
+            else if (value is DataFactoryExpression<int?> nullableIntExpression)
+                Serialize(writer, nullableIntExpression);
+            else if (value is DataFactoryExpression<double> doubleExpression)
+                Serialize(writer, doubleExpression);
+            else if (value is DataFactoryExpression<double?> nullableDoubleExpression)
+                Serialize(writer, nullableDoubleExpression);
+            else if (value is DataFactoryExpression<Array?> arrayExpression)
+                Serialize(writer, arrayExpression);
+            else if (value is DataFactoryExpression<bool> boolExpression)
+                Serialize(writer, boolExpression);
+            else if (value is DataFactoryExpression<bool?> nullableBoolExpression)
+                Serialize(writer, nullableBoolExpression);
+            else
+            {
+                Type typeToConvert = value.GetType();
+                if (IsGenericDataFactoryList(typeToConvert))
+                {
+                    string methodName = typeToConvert.GenericTypeArguments[0].GetGenericTypeDefinition() == typeof(IList<>)
+                        ? nameof(SerializeList)
+                        : nameof(SerializeReadOnlyList);
+                    var methodInfo = GetGenericSerializationMethod(typeToConvert, methodName);
+                    methodInfo!.Invoke(null, new object[] { writer, value });
+                }
+                else
+                {
+                    throw new InvalidOperationException($"Unable to convert {value.GetType().Name} into a DataFactoryExpression<T>");
+                }
+            }
+        }
+
+        private static MethodInfo GetGenericSerializationMethod(Type typeToConvert, string methodName)
+        {
+            return typeof(DataFactoryExpressionJsonConverter)
+                .GetMethod(
+                    methodName,
+                    BindingFlags.Static | BindingFlags.NonPublic)!
+                .MakeGenericMethod(typeToConvert.GenericTypeArguments[0].GenericTypeArguments[0]);
+        }
+
+        private static bool IsGenericDataFactoryList(Type type)
+        {
+            return type.IsGenericType &&
+                   type.GetGenericTypeDefinition() == typeof(DataFactoryExpression<>) &&
+                   type.GenericTypeArguments[0].IsGenericType &&
+                   IsGenericListType(type.GenericTypeArguments[0].GetGenericTypeDefinition()) &&
+                   type.GenericTypeArguments[0].GenericTypeArguments[0].GetCustomAttributes().Any(a => a.GetType() == typeof(JsonConverterAttribute));
+        }
+
+        private static bool IsGenericListType(Type type) => type == typeof(IList<>) || type == typeof(IReadOnlyList<>);
+
+        private static void Serialize<T>(Utf8JsonWriter writer, DataFactoryExpression<T?> expression)
+        {
+            if (expression.HasLiteral)
+            {
+                writer.WriteObjectValue(expression.LiteralInternal!);
```

Comment:
The nullability in WriteObjectValue is apparently incorrect - it does support nulls: https://github.com/Azure/autorest.csharp/blob/feature/v3/src/assets/Generator.Shared/Utf8JsonWriterExtensions.cs#L69